src/Storage/Factory.php

```diff
@@ -28,28 +28,29 @@ abstract class Factory
     /**
      * Create and return a StorageInterface instance
      *
-     * @param  string                             $type
-     * @param  array|Traversable                  $options
-     * @return StorageInterface
      * @throws Exception\InvalidArgumentException For unrecognized $type or individual options.
      */
-    public static function factory($type, $options = [])
+    public static function factory(?string $type, iterable $options = []): StorageInterface
```

Comment:
```suggestion
    public static function factory(string $type, array $options = []): StorageInterface
```

Make this not nullable and drop the exception below.

Also, It's probably unnecessary for `$options` to be `iterable` - Hang over from the days of `Laminas\Config` IMO. Requiring an array means we don't have to use `ArrayUtils`